File: solutions/python/largest-series-product/1/largest_series_product.py
def largest_product(series, size):
    if size > len(series):
        raise ValueError("span must be smaller than string length")
    if size < 1:
        raise ValueError("span must not be negative")
    for digit in series:
        if digit not in "0123456789":
            raise ValueError("digits input must only contain digits")
    largest = 0
    for i in range(len(series) - size + 1):
        product = int(series[i])
        for j in range(size - 1):
            product *= int(series[i + j + 1])
        if product > largest:
            largest = product
    return largest

File: solutions/lua/largest-series-product/1/largest-series-product.lua
return function(config)
    if config.span == 0 then return 1 end
    assert(config.span > 0, 'span should not be negative')
    assert(#config.digits > 0, 'string should be not null')
    assert(#config.digits >= config.span,
           'span should be not bigger than string')

    local largest_product = -math.huge
    local step = config.span - 1

    for i = 1, #config.digits - step do
        local product = 1
        for digit in config.digits:sub(i, i + step):gmatch('.') do
            product = product * digit
        end
        largest_product = math.max(largest_product, product)
    end

    return largest_product
end
